test_scripts/API/Navigation/SendLocation/007_SendLocation_invalid_types.lua

```diff
@@ -0,0 +1,111 @@
+-- Description:
+-- App sends SendLocation with invalid payload (invalid json)
+
+-- Pre-conditions:
+-- a. HMI and SDL are started
+-- b. appID is registered on SDL
+
+-- Steps:
+-- App sends SendLocation with invalid payload (invalid json)
+
+-- Expected:
+
+-- SDL validates parameters of the request
+-- SDL responds INVALID_DATA, success: false
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local commonSendLocation = require('test_scripts/API/SendLocation/commonSendLocation')
+
+--[[ Local Variables ]]
+local requestParams = {
+    longitudeDegrees = 1.1,
+    latitudeDegrees = 1.1,
+    addressLines = 
+    { 
+        "line1",
+        "line2",
+    }, 
+    address = {
+        countryName = "countryName",
+        countryCode = "countryName",
+        postalCode = "postalCode",
+        administrativeArea = "administrativeArea",
+        subAdministrativeArea = "subAdministrativeArea",
+        locality = "locality",
+        subLocality = "subLocality",
+        thoroughfare = "thoroughfare",
+        subThoroughfare = "subThoroughfare"
+    },
+    timeStamp = {
+        millisecond = 0,
+        second = 40,
+        minute = 30,
+        hour = 14,
+        day = 25,
+        month = 5,
+        year = 2017,
+        tz_hour = 5,
+        tz_minute = 30
+    },
+    locationName = "location Name",
+    locationDescription = "location Description",
+    phoneNumber = "phone Number",
+    deliveryMode = "PROMPT",
+    locationImage = 
+    { 
+        value = "icon.png",
+        imageType = "DYNAMIC",
+    }
+}
+
+--[[ Local Functions ]]
+local function sendLocation(params, self)
+    local cid = self.mobileSession1:SendRPC("SendLocation", params)
+
+    EXPECT_HMICALL("Navigation.SendLocation", params):Times(0)
+
+    self.mobileSession1:ExpectResponse(cid, {success = false, resultCode = "INVALID_DATA"})
+
+    commonSendLocation.delayedExp()
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", commonSendLocation.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", commonSendLocation.start)
+runner.Step("RAI, PTU", commonSendLocation.registerApplicationWithPTU)
+runner.Step("Activate App", commonSendLocation.activateApp)
+
+for key,value in pairs(requestParams) do
+    local parametersToSend = requestParams
+    local function replaceType(val)
+        if (type(val) == "string") then
+            return 5
+        else
+            return "5"
+        end
+    end
+    if (type(value) == "table") then
+        for subKey,subValue in pairs(value) do
+            parametersToSend[key][subKey] = replaceType(subValue)
+            runner.Title("Test")
+            runner.Step("SendLocation-invalid-type-of-" .. tostring(subKey), sendLocation, {parametersToSend})
+        end
+    else
+        parametersToSend[key] = replaceType(subValue)
```

Comment:
@Itileda Variable 'subValue' is undefined

Reply:
@dboltovskyi , fixed in 6f2e7484bc3d0171c93beefc946839185e141366